src/Processors/Transforms/TTLTransform.cpp

```diff
@@ -131,6 +169,25 @@ void TTLTransform::consume(Chunk chunk)
     removeSpecialColumnRepresentations(chunk);
     auto block = getInputPort().getHeader().cloneWithColumns(chunk.detachColumns());
 
+    /// Fill expired columns with default values which will later be handled in TTLColumnAlgorithm
+    for (const auto & [column, data] : expired_columns_data)
+    {
+        auto default_column
+            = ITTLAlgorithm::executeExpressionAndGetColumn(data.default_expression, block, data.default_column_name);
+        if (default_column)
+            default_column = default_column->convertToFullColumnIfConst();
+        else
+            default_column = data.type->createColumnConstWithDefaultValue(block.rows())->convertToFullColumnIfConst();
```

Comment:
Do I assume correctly that the case of `data.default_expression == nullptr` is handled here?